src/robotFactory.js

```diff
@@ -1,10 +1,95 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(
+    name,
+    weight,
+    coords = {},
```

Comment:
```suggestion
    coords,
```

Coords are always an object. No need to assign it to an empty object.